exchanges/apis/kucoin.py

```diff
@@ -24,14 +35,37 @@ def brequest(
         api_path = f"/{self.api_prefix}/v{api_version}/{endpoint}"
         headers = self.DEFAULT_HEADERS.copy()
 
-        data = data or {}
-
-        # TODO: implement authentication for writes if needed
         if authenticate:
-            pass
+            headers.update(self.auth_headers(api_version, method, api_path, data))
 
         url = base_url + api_path
-        return self.request(url, method, params, data, headers)
+
+        limiter = RateLimiter(
+            max_calls=100,
+            period=10,
```

Comment:
Should be constants :)

Reply:
Hmm, they aren't constants in Bitfinex. I'll see if there's a better way also given that rates are limited by endpoint